autogpt_platform/backend/backend/blocks/code_executor_test.py

```
@@ -101,7 +101,28 @@ def test_malicious_value_cannot_break_out_of_code_channel(self):
         envs, prefix = build_variable_injection(variables, ProgrammingLanguage.PYTHON)
         # The dangerous string lives only in the env payload, never in the code.
         assert "os.system" not in prefix
-        assert envs[VARIABLES_ENV_KEY] == json.dumps(variables)
+        assert envs[VARIABLES_ENV_KEY] == json.dumps(variables, ensure_ascii=True)
+
+    def test_unicode_and_emojis_serialize_safely_as_ascii(self):
+        """Emojis and UTF-8 characters serialize safely to ASCII to prevent env var encoding errors."""
+        variables = {"greeting": "Holidays 🎅🏻", "name": "日本語"}
+        envs, prefix = build_variable_injection(variables, ProgrammingLanguage.PYTHON)
```

Ruff (0.16.2)

[warning] Unpacked variable `prefix` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)